### PR TITLE
Adding dmgbuild

### PR DESCRIPTION
**Pull request by marcelotduarte**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/dmgbuild/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 7ced2603d684e29c22b4cd507d1e15a1907e91b86259924b8cfe480d80553b43
+
+build:
+  number: 0
+  skip: true  # [not osx]
```
Comment (ocefpaf) [inline recipes/dmgbuild/meta.yaml:14]: If this is a macOS only tool we should make it a OS-noarch package. See https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies

Comment (ocefpaf) [reply recipes/dmgbuild/meta.yaml]: You will need to add the conda-forge.yml file. Please read the link above.

File: recipes/dmgbuild/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{% set name = "dmgbuild" %}
+{% set version = "1.6.1" %}
+{% set sha256 = "7ced2603d684e29c22b4cd507d1e15a1907e91b86259924b8cfe480d80553b43" %}
```
[the remark this answers is not in the document]
Comment (marcelotduarte) [reply recipes/dmgbuild/meta.yaml:3]: Can I use only name and version?

[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/dmgbuild/meta.yaml:3]: Yep.

File: recipes/dmgbuild/meta.yaml (hunk 1 above)
Comment (marcelotduarte) [reply recipes/dmgbuild/meta.yaml]: Please check if it is correct.

File: recipes/dmgbuild/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 7ced2603d684e29c22b4cd507d1e15a1907e91b86259924b8cfe480d80553b43
+
+build:
+  number: 0
```
Comment (ocefpaf) [inline recipes/dmgbuild/meta.yaml:13]:
```suggestion
  number: 0
  noarch: python
```

File: recipes/dmgbuild/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  entry_points:
+    - dmgbuild = dmgbuild.__main__:main
+
+requirements:
+  host:
+    - python
```
Comment (ocefpaf) [inline recipes/dmgbuild/meta.yaml:20]:
```suggestion
    - python >=3.9
```

File: recipes/dmgbuild/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - pip
+    - setuptools
+  run:
+    - python
```
Comment (ocefpaf) [inline recipes/dmgbuild/meta.yaml:24]:
```suggestion
    - python >=3.9
```

File: recipes/dmgbuild/meta.yaml (hunk 3 above)
Comment (marcelotduarte) [reply recipes/dmgbuild/meta.yaml:13]: When I used, this caused a warning https://github.com/conda-forge/staged-recipes/pull/26775#issuecomment-2193968539

Comment (ocefpaf) [reply recipes/dmgbuild/meta.yaml:13]: B/c you did not had the conda-forge.yml before.

File: recipes/dmgbuild/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    - dmgbuild = dmgbuild.__main__:main
+
+requirements:
+  host:
+    - python >=3.9
+    - pip
+    - setuptools
+  run:
+    - python >=3.9
+    - importlib_resources  # [py<39]
```
Comment (ocefpaf) [inline recipes/dmgbuild/meta.yaml:26]: If we are building py>=39 we don't need this. If you do want to build for py38 you can change the constraints above but remove the selector here, `importlib_resources` is just a stub in py>=3.9, so it is fine to install it there.

```suggestion
```

**Comment by marcelotduarte** [comment]
@ocefpaf skip is used in various packages, an example: https://github.com/conda-forge/pywin32-ctypes-feedstock/blob/main/recipe/meta.yaml#L13

File: recipes/dmgbuild/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    - dmgbuild = dmgbuild.__main__:main
+
+requirements:
+  host:
+    - python >=3.9
+    - pip
+    - setuptools
+  run:
+    - python >=3.9
+    - __osx                # [osx]
```
Comment (ocefpaf) [inline recipes/dmgbuild/meta.yaml:26]: I wonder if there is a linter bug here. This should be enough. @isuruf, am I missing something obvious here?

**Comment by ocefpaf** [comment]
> @ocefpaf skip is used in various packages, an example: https://github.com/conda-forge/pywin32-ctypes-feedstock/blob/main/recipe/meta.yaml#L13

In various old packages, before we implemented OS-noarch, yes.

File: recipes/dmgbuild/meta.yaml (hunk 7 above)
Comment (marcelotduarte) [reply recipes/dmgbuild/meta.yaml]: @ocefpaf I found and followed what was made in https://github.com/conda-forge/applaunchservices-feedstock/blob/main/recipe/meta.yaml but it still fails.
There really must be an error in the linter.
One possibility is that the conda-forge.yml file is not being considered, as I renamed it to .yaml and it continues to give the same error.
Can you commit and then we can make any necessary adjustments to the feedstock?

Comment (marcelotduarte) [reply recipes/dmgbuild/meta.yaml]: @ocefpaf Thanks!
https://github.com/conda-forge/dmgbuild-feedstock/pull/1

**Pull request merged**